internal/security/anchore_client.go

```diff
@@ -180,16 +190,25 @@ func (a anchoreClient) DeleteAccount(ctx context.Context, accountName string) er
 	a.logger.Info("deleting anchore account", fnCtx)
 
 	// update the status of the account before delete
-	s, ur, err := a.getRestClient().UserManagementApi.UpdateAccountState(a.authorizedContext(ctx), accountName, anchore.AccountStatus{State: "disabled"})
+	var desiredState anchore.AccountStatus
+	disabled := "disabled"
+	desiredState.State = &disabled
+
+	updateAccountRequest := a.getRestClient().UserManagementApi.UpdateAccountState(a.authorizedContext(ctx), accountName)
+	updateAccountRequest.DesiredState(desiredState)
 
-	if err != nil || ur.StatusCode != http.StatusOK || s.State != "disabled" {
+	s, ur, err := a.getRestClient().UserManagementApi.UpdateAccountStateExecute(updateAccountRequest)
+
+	if err != nil || ur.StatusCode != http.StatusOK || s.State != &disabled {
```

Comment:
**Question**: Is it intentional we are comparing the state pointers instead of their values? I would expect the pointers to mismatch even when they are pointing to the same values but this request execution could have a custom behavior implemented.

Reply:
Thanks for catching this, it would indeed make more sense!